aries_vcx_core/src/anoncreds/credx_anoncreds.rs

```diff
@@ -56,14 +60,144 @@ pub struct RevocationRegistryInfo {
     pub used_ids: HashSet<u32>,
 }
 
+/// Adapter used so that credx does not depend strictly on the vdrtools-wallet
+/// Will get removed when the wallet and anoncreds interfaces are de-coupled.
+#[derive(Debug)]
+struct WalletAdapter(Arc<dyn BaseWallet>);
```

Comment:
im not sure i understand. i thought credx anoncreds would already be decoupled via `Arc<dyn BaseWallet>`

Reply:
The long-term plan is to remove the interdependencies between components so that, let's say, some `Anoncreds` implementation does not contain a `Wallet` and vice-versa. 

I temporarily added this so that the `BaseWallet` implementation is close to the actual type and can get removed together. The alternative was to keep the `Arc<dyn BaseWallet>` implementation of the `BaseWallet` trait, but that's too generic and easily forgettable.